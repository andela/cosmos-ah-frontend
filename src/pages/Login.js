import React, { useState } from 'react';
import { Button, Form, Message } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { signin } from '../state/auth/actions';

const Signin = props => {
  const [formData, setFormData] = useState({
    email: '',
    password: ''
  });
  const [errors, setErrors] = useState({});
  const [loading, setLoading] = useState(false);

  const handleSubmit = async event => {
    try {
      event.preventDefault();
      setLoading(true);
      await props.signin({ email, password });
      props.history.push('/dashboard');
    } catch (error) {
      if (error.response.data.data.password) {
        setErrors(() => ({ ...error.response.data.data }));
        setLoading(false);
      } else {
        setErrors(() => ({ ...error.response.data }));
        setLoading(false);
      }
    }
  };

  const handleChange = event => {
    event.persist();
    setFormData(() => ({ ...formData, [event.target.name]: event.target.value }));
  };

  const { email, password } = formData;
  return (
    <Form onSubmit={handleSubmit} loading={loading}>
    <h1>Welcome to the Sign In page</h1>
    {errors.password && (
          <Message negative>
            <Message.Header>Something went wrong</Message.Header>
            <p>{errors.password}</p>
          </Message>
    )}
    {errors.data && (
          <Message negative>
            <Message.Header>Something went wrong</Message.Header>
            <p>{errors.data}</p>
          </Message>
    )}
      <Form.Field>
        <label htmlFor='email'>Email</label>
        <input type='email' id='email' name='email' placeholder='example@example.com' onChange={handleChange} value={email} required/>
      </Form.Field>
      <Form.Field>
        <label htmlFor='password'>Password</label>
        <input type='password' id='password' name='password' placeholder='your password...' onChange={handleChange} value={password} required/>
      </Form.Field>
      <Button primary>Login</Button>
    </Form>
  );
};

export default connect(null, { signin })(Signin);
