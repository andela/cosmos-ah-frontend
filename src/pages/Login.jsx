import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import {
  Form, Grid, Header, Message
} from 'semantic-ui-react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import SocialButton from '../components/SocialMediaButton';
import { ButtonComponent } from '../components/Button';
import { loginAction } from '../state/auth/actions';
import AppLogo from '../components/AppLogo';
import logo from '../assets/images/logo.png';

const Body = styled.div`
  display: flex;
  height: 100vh;
  flex-direction: column;
  color: #333;
`;

const Logo = styled.div`
  margin: 25px;
  @media (max-width: 540px) {
    text-align: center;
    img {
      width: 100px;
    }
  }
`;

const Wrap = styled.div`
width: 500px;
&&& {
  input {
    background-color: #E8E8E8 !important;
    border-radius: 10px !important;
    border: none !important;
  }
  label {
    color: #3A8FDD !important;
    }
}
  @media (max-width: 500px) {
    max-width: 400px;
  }
  @media (max-width: 400px) {
    max-width: 300px;
  }
`;

const Space = styled.div`
  margin: 20px;
`;

const StyledLink = styled.p`
  text-align: right;
`;

const Center = styled.div`
  margin: 0 auto;
  text-align: center;
`;


const transformObjectValToString = obj => {
  let data = [];
  if (obj) {
    data = [...Object.values(obj)];
    return data;
  }
  return data[0];
};

const Login = props => {
  const { loginState } = props;
  const validatorErrors = transformObjectValToString(loginState.data);
  const [formData, setFormData] = useState({
    email: '',
    password: ''
  });

  const handleSubmit = event => {
    event.persist();
    event.preventDefault();
    props.loginAction({ email, password });
  };

  const handleChange = event => {
    event.persist();
    setFormData(() => ({ ...formData, [event.target.name]: event.target.value }));
  };

  if (loginState.id) {
    props.history.push('/profile');
  }

  const { email, password } = formData;

  return (
  <Body>
    <Logo>
      <AppLogo logo={logo} />
    </Logo>
    <Center>
      <Wrap>
        <Grid.Column>
          <Center>
            <Header as='h1' color='blue' size='huge'>Login</Header>
          </Center>
          <SocialButton />
          <Header color='blue'>
            <h4>or login using your email address</h4>
          </Header>
          <Form onSubmit={handleSubmit} loading={loginState.loadingState}>
            <Form.Input size='big' icon={{ name: 'envelope outline', color: 'blue' }} iconPosition='left' placeholder='Email Address' name='email' onChange={handleChange} value={email} required={true}/>
            <Form.Input size='big' icon={{ name: 'lock', color: 'blue' }} iconPosition='left' placeholder='Password' type='password' name='password' onChange={handleChange} value={password} required={true}/>
            {(validatorErrors !== undefined) && (
            <Message negative>
              <p>{validatorErrors}</p>
            </Message>
            )}
            <StyledLink><Link to='/forgot-password'>Forgot password?</Link></StyledLink>
            <ButtonComponent color='blue' size='big'>
              Continue
            </ButtonComponent>
          </Form>
          <Space>
            <Grid centered>
            <Center>
              <Header>
                <h3>Don't have an account yet? <Link to='/signup'>Create One</Link></h3>
            </Header>
            </Center>
            </Grid>
          </Space>
        </Grid.Column>
      </Wrap>
    </Center>
  </Body>
  );
};

function mapStateToProps(state) {
  return { loginState: state.Auth.login };
}

export default connect(mapStateToProps, { loginAction })(Login);
