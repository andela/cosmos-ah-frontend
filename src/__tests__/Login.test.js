import React from 'react';
import { render, cleanup, getByText } from 'react-testing-library';
import Login from '../pages/Login';

afterEach(cleanup);

test('Renders <Login />', () => {
  const LoginComponent = render(<Login />);
  expect(LoginComponent).toBeTruthy();
  expect(LoginComponent).toMatchSnapshot();
});

test('Renders <Login /> form', () => {
  const { getByPlaceholderText } = render(<Login />);
  expect(getByPlaceholderText('Email Address')).toBeTruthy();
  expect(getByPlaceholderText('Password')).toBeTruthy();
});


test('render <Login />  Login Header', () => {
  const LoginHeader = render(<Login />);
  const HeaderText = LoginHeader.getByText('Login');
  expect(HeaderText.textContent).toBe('Login');
});
