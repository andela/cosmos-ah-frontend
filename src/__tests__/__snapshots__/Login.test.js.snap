// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Login /> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bwzfXH jtCEHj"
      >
        <div
          class="sc-htpNat iSSvoI"
        >
          <a
            href="/"
          >
            <img
              src="https://res.cloudinary.com/drmrayjr0/image/upload/v1557246832/Screenshot_2019-05-07_at_5.33.23_PM_p7frml.png"
              width="150"
            />
          </a>
        </div>
        <div
          class="sc-bZQynM bYmjcT"
        >
          <div
            class="sc-bxivhb dAJZi"
          >
            <div
              class="column"
            >
              <div
                class="sc-bZQynM bYmjcT"
              >
                <h1
                  class="ui blue huge header"
                >
                  Login
                </h1>
              </div>
              <div
                class="sc-bdVaJa cLnXBB"
              >
                <div
                  class="ui centered grid"
                >
                  <button
                    class="ui blue massive basic circular icon button"
                    data-testid="facebook"
                  >
                    <i
                      aria-hidden="true"
                      class="facebook icon"
                    />
                  </button>
                  <button
                    class="ui red massive basic circular icon button"
                    data-testid="google"
                  >
                    <i
                      aria-hidden="true"
                      class="google plus icon"
                    />
                  </button>
                  <button
                    class="ui blue massive basic circular icon button"
                    data-testid="twitter"
                  >
                    <i
                      aria-hidden="true"
                      class="twitter icon"
                    />
                  </button>
                  <button
                    class="ui blue massive basic circular icon button"
                    data-testid="linkedin"
                  >
                    <i
                      aria-hidden="true"
                      class="linkedin icon"
                    />
                  </button>
                </div>
              </div>
              <div
                class="ui blue header"
              >
                <h4>
                  or login using your email address
                </h4>
              </div>
              <form
                class="ui form"
              >
                <div
                  class="field"
                >
                  <div
                    class="ui big left icon input"
                  >
                    <input
                      placeholder="Email Address"
                      type="text"
                    />
                    <i
                      aria-hidden="true"
                      class="blue envelope outline icon"
                    />
                  </div>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="ui big left icon input"
                  >
                    <input
                      placeholder="Password"
                      type="password"
                    />
                    <i
                      aria-hidden="true"
                      class="blue lock icon"
                    />
                  </div>
                </div>
                <p
                  class="sc-EHOje dhYIww"
                >
                  <a
                    href="#"
                  >
                    Forgot password?
                  </a>
                </p>
                <div>
                  <button
                    class="ui blue big fluid button"
                  >
                    Continue
                  </button>
                </div>
              </form>
              <div
                class="sc-ifAKCX bdgWuN"
              >
                <div
                  class="ui centered grid"
                >
                  <div
                    class="sc-bZQynM bYmjcT"
                  >
                    <div
                      class="ui header"
                    >
                      <h3>
                        Don't have an account yet? 
                        <a
                          href="#"
                        >
                          Create One
                        </a>
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bwzfXH jtCEHj"
    >
      <div
        class="sc-htpNat iSSvoI"
      >
        <a
          href="/"
        >
          <img
            src="https://res.cloudinary.com/drmrayjr0/image/upload/v1557246832/Screenshot_2019-05-07_at_5.33.23_PM_p7frml.png"
            width="150"
          />
        </a>
      </div>
      <div
        class="sc-bZQynM bYmjcT"
      >
        <div
          class="sc-bxivhb dAJZi"
        >
          <div
            class="column"
          >
            <div
              class="sc-bZQynM bYmjcT"
            >
              <h1
                class="ui blue huge header"
              >
                Login
              </h1>
            </div>
            <div
              class="sc-bdVaJa cLnXBB"
            >
              <div
                class="ui centered grid"
              >
                <button
                  class="ui blue massive basic circular icon button"
                  data-testid="facebook"
                >
                  <i
                    aria-hidden="true"
                    class="facebook icon"
                  />
                </button>
                <button
                  class="ui red massive basic circular icon button"
                  data-testid="google"
                >
                  <i
                    aria-hidden="true"
                    class="google plus icon"
                  />
                </button>
                <button
                  class="ui blue massive basic circular icon button"
                  data-testid="twitter"
                >
                  <i
                    aria-hidden="true"
                    class="twitter icon"
                  />
                </button>
                <button
                  class="ui blue massive basic circular icon button"
                  data-testid="linkedin"
                >
                  <i
                    aria-hidden="true"
                    class="linkedin icon"
                  />
                </button>
              </div>
            </div>
            <div
              class="ui blue header"
            >
              <h4>
                or login using your email address
              </h4>
            </div>
            <form
              class="ui form"
            >
              <div
                class="field"
              >
                <div
                  class="ui big left icon input"
                >
                  <input
                    placeholder="Email Address"
                    type="text"
                  />
                  <i
                    aria-hidden="true"
                    class="blue envelope outline icon"
                  />
                </div>
              </div>
              <div
                class="field"
              >
                <div
                  class="ui big left icon input"
                >
                  <input
                    placeholder="Password"
                    type="password"
                  />
                  <i
                    aria-hidden="true"
                    class="blue lock icon"
                  />
                </div>
              </div>
              <p
                class="sc-EHOje dhYIww"
              >
                <a
                  href="#"
                >
                  Forgot password?
                </a>
              </p>
              <div>
                <button
                  class="ui blue big fluid button"
                >
                  Continue
                </button>
              </div>
            </form>
            <div
              class="sc-ifAKCX bdgWuN"
            >
              <div
                class="ui centered grid"
              >
                <div
                  class="sc-bZQynM bYmjcT"
                >
                  <div
                    class="ui header"
                  >
                    <h3>
                      Don't have an account yet? 
                      <a
                        href="#"
                      >
                        Create One
                      </a>
                    </h3>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
